### ISSUE TITLE
Cannot apply plugin into Android multi module project

### ISSUE THREAD

**Issue by mpowloka** [JLLeitschuh/ktlint-gradle#174]
I cannot find a way to apply this plugin into my Android project. I tried both approaches:

```
plugins {
    id "org.jlleitschuh.gradle.ktlint" version "6.3.0"
}
```
and 

```
buildscript {
  repositories {
    maven {
      url "https://plugins.gradle.org/m2/"
    }
  }
  dependencies {
    classpath "gradle.plugin.org.jlleitschuh.gradle:ktlint-gradle:6.3.0"
  }
}

apply plugin: "org.jlleitschuh.gradle.ktlint"
```

Also I tried to apply it in multiple gradle files (root file and module files). I keep getting following error:

```
Unable to load class 'org.gradle.api.provider.SetProperty'.
Possible causes for this unexpected error include:<ul><li>Gradle's dependency cache may be corrupt (this sometimes occurs after a network connection timeout.)
Re-download dependencies and sync project (requires network)</li><li>The state of a Gradle build process (daemon) may be corrupt. Stopping all Gradle daemons may solve this problem.
Stop Gradle build processes (requires restart)</li><li>Your project may be using a third-party plugin which is not compatible with the other plugins in the project or the version of Gradle requested by the project.</li></ul>In the case of corrupt Gradle processes, you can also try closing the IDE and then killing all Java processes.
```

Any of solutions proposed by error didn't help in any way.

Gradle wrapper version is 4.4 and gradle plugin version is 3.0.1.
I am willing to provide any further required information.

**Comment by Tapchicoma** [comment]
With Android you should use the second way.

Also the id of the plugin has changed, due to recent Gradle plugin portal changes and now it is "org.jlleitschuh.gradle:ktlint-gradle:6.3.0".

See #172.

**Comment by mpowloka** [comment]
Hello @Tapchicoma ,

I updated my gradle file to use:

```
buildscript {
    repositories {
        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }
    dependencies {
        classpath "org.jlleitschuh.gradle:ktlint-gradle:6.3.0"
    }
}

apply plugin: "org.jlleitschuh.gradle.ktlint"
```

but I keep getting the very same error...

**Comment by Tapchicoma** [comment]
Can you show your project repo?

If not, can you run: `./gradlew ktlintCheck -s` and paste here stacktrace?

**Comment by Tapchicoma** [comment]
Ah, found the problem - https://docs.gradle.org/current/javadoc/org/gradle/api/provider/SetProperty.html - it is only available since Gradle 4.5

Possible solutions:
- Update Android plugin to 3.2 and Gradle to 4.5 version
- Use older ktlint-gradle version `5.1.0` - this should work in your setup

@JLLeitschuh I am tending not to fix this, but update minimal supported Gradle version. Also really need to do #148 that will run against lower supported and latest released Gradle versions to avoid such issues in the future.

**Comment by mpowloka** [comment]
Hey,

I downgraded to 5.1 at it looks like it is working. Unfortunately I had to leave my workplace so I don't know if 6.3 will work with gradle 4.5, but I will let you know tomorrow morning.

Thank you very much for help!

**Comment by JLLeitschuh** [comment]
@Tapchicoma I agree about updating the minimum supported version of Grade.

**Comment by Tapchicoma** [comment]
Interesting stacktrace, I don't see anything related to `ktlint-gradle`.

In versions `6.x.x` we removed usages of `afterEvaluate {}` and this line `java.lang.ClassNotFoundException: com.android.ide.common.res2.ResourceSet` related to android plugin itself.
